src/untangled/client/impl/application.cljc

```diff
@@ -218,19 +218,21 @@
                                       (swap! initial-state assoc :ui/locale "en-US")
                                       initial-state)
                                     (assoc initial-state :ui/locale "en-US"))
-        config                    {:state       initial-state-with-locale
-                                   :send        (fn [tx cb]
-                                                  (server-send (assoc app :reconciler @rec-atom) tx cb))
-                                   :migrate     (or migrate tempid-migrate)
-                                   :normalize   true
-                                   :remotes     remotes
-                                   :pathopt     pathopt
-                                   :merge-ident (fn [reconciler app-state ident props]
-                                                  (update-in app-state ident (comp sweep-one merge) props))
-                                   :merge-tree  (fn [target source]
-                                                  (merge-handler mutation-merge target source))
-                                   :parser      parser
-                                   :shared      shared}
+        reconciler-options        (update reconciler-options :migrate #(or % tempid-migrate))
+        config                    (merge {:state       initial-state-with-locale
+                                          :send        (fn [tx cb]
+                                                         (server-send (assoc app :reconciler @rec-atom) tx cb))
+                                          :migrate     nil
+                                          :normalize   true
+                                          :remotes     remotes
+                                          :pathopt     true
+                                          :merge-ident (fn [reconciler app-state ident props]
+                                                         (update-in app-state ident (comp sweep-one merge) props))
+                                          :merge-tree  (fn [target source]
+                                                         (merge-handler mutation-merge target source))
+                                          :parser      parser
+                                          :shared      nil}
+                                    reconciler-options)
```

Comment:
I understand that in most cases people will not want to override this, but don't you think the user should ultimately have the power to do what he seems fit? What if I want to extend some of those behaviors, I always would like to have the option to, especially on something so critical to Om as the reconciler. Maybe we add documentation saying about "if you are going to override this, do it at your risk". Preventing this is like locking the user on the cases you are preventing, instead of letting it open to the user to extend.